### PR TITLE
Vulkan: support depth frame buffer

### PR DESCRIPTION
**Pull request by jaebaek**
Related to #206

### CODE REVIEW

File: tests/cases/draw_triangle_list_with_depth.amber [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 The Amber Authors.
```
Comment (dj2) [inline tests/cases/draw_triangle_list_with_depth.amber]: 2019

File: tests/cases/draw_triangle_list_with_depth.amber [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+[fragment shader]
+#version 430
+
+layout(location = 0) in vec4 frag_color;
+layout(location = 0) out vec4 final_color;
+
+void main() {
+  final_color = frag_color;
```
Comment (dj2) [inline tests/cases/draw_triangle_list_with_depth.amber]: This doesn't appear to use the depth buffer? How do you know if it's working?

Comment (dj2) [reply tests/cases/draw_triangle_list_with_depth.amber]: Can you add a comment in here that that's what's happening?

Comment (jaebaek) [reply tests/cases/draw_triangle_list_with_depth.amber]: done

**Pull request merged**